packages/containers/src/List/utils.js

```diff
@@ -0,0 +1,101 @@
+import cmf from '@talend/react-cmf';
+import get from 'lodash/get';
+import { Map, List } from 'immutable';
+
+function contains(listItem, query, columns) {
+	let item = listItem;
+	if (Map.isMap(listItem)) {
+		item = listItem.toJS();
+	}
+	for (const column of columns) {
+		if (typeof item[column.key] === 'string') {
+			if (item[column.key].toLowerCase().indexOf(query.toLowerCase()) !== -1) {
+				return true;
+			}
+		}
+	}
+	return false;
+}
+
+function getCollection(state, collectionId) {
+	return state.cmf.collections.get(collectionId);
+}
+
+export function getCollectionItems(state, collectionId) {
+	const collection = getCollection(state, collectionId);
+
+	if (Map.isMap(collection)) {
+		return collection.get('items');
+	}
+	return collection;
+}
+
+export function configureGetPagination(state, { collectionId }) {
+	const collection = getCollection(state, collectionId);
+
+	if (Map.isMap(collection)) {
+		return collection.get('pagination');
+	}
+
+	return null;
+}
+
+export function getComponentState(collectionId) {
+	return state => state.cmf.components.getIn(['Container(List)', collectionId || 'default']);
+}
+
+export function sortList(componentState, localConfig, items) {
+	if (!List.isList(items)) {
+		return new List();
+	}
+	let results = items;
+	if (componentState) {
+		const sortBy = componentState.get('sortOn');
+		const sortAsc = componentState.get('sortAsc');
```

Comment:
Those values will be given in specifics args.
And the exist check of componentState will not be needed anymore